docs/topics/signals.rst

```diff
@@ -52,9 +52,58 @@ Deferred signal handlers
 ========================
 
 Some signals support returning :class:`~twisted.internet.defer.Deferred`
-objects from their handlers, see the :ref:`topics-signals-ref` below to know
-which ones.
+objects from their handlers, allowing you to run asynchronous code that
+does not block Scrapy. If a signal handler returns a
+:class:`~twisted.internet.defer.Deferred`, Scrapy waits for that
+:class:`~twisted.internet.defer.Deferred` to fire.
 
+Let's take an example:
+::
+
+    class SignalSpider(scrapy.Spider):
+        name = 'signals'
+        start_urls = ['http://quotes.toscrape.com/page/1/']
+
+        @classmethod
+        def from_crawler(cls, crawler, *args, **kwargs):
+            spider = super(SignalSpider, cls).from_crawler(crawler, *args, **kwargs)
+            crawler.signals.connect(spider.item_scrapped, signal=signals.item_scraped)
+            return spider
+
+        def item_scrapped(self, item):
+            # Send the scrapped item to the server
+            dfd = treq.post(
+                'http://localhost:3000/add',
+                json.dumps({
+                    'text': item['text'],
+                    'author': item['author'],
+                    'tags': item['tags'],
+                }).encode('ascii'),
+                headers={b'Content-Type': [b'application/json']}
+            )
+            dfd.addCallback(print_response)
+
+            # Next item will be scrapped only after dfd is fired
+            # As items are processed in parallel many dfd will
+            # be fired at the same time (refer settings['CONCURRENT_ITEMS'])
+            # hence, next batch of items will be processed after all
+            # dfd's are fired
```

Comment:
```suggestion
            # The next item will be scraped only after the deferred (d)
            # is fired. Because items are processed in parallel by batches
            # (see the CONCURRENT_ITEMS setting), many deferreds
            # will be fired at the same time, but the next batch of items will
            # only start being processed after all deferreds have been fired.
```

Reply:
I have remove this note and updated to
```python
def item_scraped(self, item):
    # Send the scraped item to the server
    d = treq.post(
        'http://localhost:3000/add',
        json.dumps({
            'text': item['text'],
            'author': item['author'],
            'tags': item['tags'],
        }).encode('ascii'),
        headers={b'Content-Type': [b'application/json']}
    )
    d.addCallback(print_response)

    # The next item will be scraped only after
    # deferred (d) is fired
    return d
```